### PR TITLE
SUBMARINE-276. Submarine runtime class config should be defined in the new configuration class.

### PR DESCRIPTION
**Pull request by yuanzac**
### What is this PR for?
There are two SubmarineConfiguration classes in the submarine project. 

Let's combine them into the new configuration class org.apache.submarine.commons.utils.SubmarineConfiguration.

### What type of PR is it?
Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-276

### How should this be tested?
https://travis-ci.org/yuanzac/hadoop-submarine/builds/611726441?utm_medium=notification&utm_source=github_status

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? Yes
* Does this needs documentation? Yes

### CODE REVIEW

File: submarine-server/server-submitter/submitter-yarn/src/main/java/org/apache/submarine/server/submitter/yarn/YarnUtils.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 import java.util.List;
 import java.util.Map;
 
-import static org.apache.submarine.commons.runtime.conf.SubmarineConfiguration.SUBMARINE_RUNTIME_APP_TYPE;
+import static org.apache.submarine.commons.utils.SubmarineConfiguration.SUBMARINE_RUNTIME_APP_TYPE;
```
Comment (liuxunorg) [inline submarine-server/server-submitter/submitter-yarn/src/main/java/org/apache/submarine/server/submitter/yarn/YarnUtils.java]: maybe `SUBMARINE_RUNTIME_CLASS` is better?

File: submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/FileSystemOperations.java [hunk 2]
```diff
@@ -179,9 +179,9 @@ public void validFileSize(String uri) throws IOException {
     } else {
       actualSizeByte = FileUtil.getDU(new File(uri));
     }
-    long maxFileSizeMB = submarineConfig
-        .getLong(SubmarineConfiguration.LOCALIZATION_MAX_ALLOWED_FILE_SIZE_MB,
-            SubmarineConfiguration.DEFAULT_MAX_ALLOWED_REMOTE_URI_SIZE_MB);
+    long maxFileSizeMB = submarineConfig.getLong(
+        SubmarineConfiguration.ConfVars.
+            SUBMARINE_LOCALIZATION_MAX_ALLOWED_FILE_SIZE_MB);
```
Comment (liuxunorg) [inline submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/FileSystemOperations.java]: Not need `DEFAULT_MAX_ALLOWED_REMOTE_URI_SIZE_MB` ?

File: docs/helper/QuickStart.md [hunk 3]
```diff
@@ -47,15 +47,15 @@ Note that if you want to quickly try Submarine on new or existing YARN cluster,
...
+Once the applicable runtime is chosen and environment is ready, a `submarine-site.xml` need to be created under  `$HADOOP_CONF_DIR`. To use the TonY runtime, please set below value in the submarine configuration.
 
 |Configuration Name | Description |
 |:---- |:---- |
 | `submarine.runtime.class` | "org.apache.submarine.server.submitter.yarn.YarnRuntimeFactory" or "org.apache.submarine.server.submitter.yarnservice.YarnServiceRuntimeFactory" |
 
 <br />
 
-A sample `submarine.xml` is here:
+A sample `submarine-site.xml` is here:
```
[the remark this answers is not in the document]
Comment (yuanzac) [reply docs/helper/QuickStart.md]: Done

File: submarine-client/src/main/java/org/apache/submarine/client/cli/AbstractCli.java [hunk 4]
```diff
@@ -42,11 +42,11 @@ public abstract int run(String[] args)
...
   @Override
   public void setConf(Configuration conf) {
-    clientContext.setSubmarineConfig(conf);
+    clientContext.setConfiguration(conf);
   }
 
   @Override
   public Configuration getConf() {
-    return clientContext.getSubmarineConfig();
+    return clientContext.getYarnConfig();
```
[the remark this answers is not in the document]
Comment (yuanzac) [reply submarine-client/src/main/java/org/apache/submarine/client/cli/AbstractCli.java]: The method getConf and setConf is defined in Tool class, which is from Hadoop code.
So we can't change them to getConfiguration or getYarnConfiguration.

File: submarine-server/server-submitter/submitter-yarn/src/main/java/org/apache/submarine/server/submitter/yarn/YarnUtils.java (hunk 1 above)
Comment (yuanzac) [reply submarine-server/server-submitter/submitter-yarn/src/main/java/org/apache/submarine/server/submitter/yarn/YarnUtils.java]: SUBMARINE_RUNTIME_APP_TYPE is another configuration. It defines the application type in the yarn. With it, Yarn UI show job type as submarine not tony.
This configuration is added in #24

File: submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/FileSystemOperations.java (hunk 2 above)
Comment (yuanzac) [reply submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/FileSystemOperations.java]: Yeah, add a method in SubmarineConfiguration:
```
public long getLong(ConfVars c) {
    return getLong(c.name(), c.getVarName(), c.getLongValue());
  }
```
This method will return the default value if the value is empty.

File: dev-support/mini-submarine/README.md [hunk 5]
```diff
@@ -247,7 +247,7 @@ cd && cd spark-script && ./run_spark.sh
 
 1. Submarine package name error
 
-   Because the package name of submarine 0.3.0 or higher has been changed from `apache.hadoop.yarn.submarine` to `apache.submarine`, So you need to set the Runtime settings in the `/usr/local/hadoop/etc/hadoop/submarine.xml` file.
+   Because the package name of submarine 0.3.0 or higher has been changed from `apache.hadoop.yarn.submarine` to `apache.submarine`, So you need to set the Runtime settings in the `/usr/local/hadoop/etc/hadoop/submarine-site.xml` file.
```
Comment (liuxunorg) [inline dev-support/mini-submarine/README.md]: `/usr/local/hadoop/etc/hadoop/submarine-site.xml` path and `submarine-current/conf/submarine-site.xml` path different?

File: submarine-client/src/main/java/org/apache/submarine/client/cli/AbstractCli.java (hunk 4 above)
Comment (liuxunorg) [inline submarine-client/src/main/java/org/apache/submarine/client/cli/AbstractCli.java]: `setConfiguration()` and `getYarnConfig()`, Why are these two functions inconsistent?

**Comment by yuanzac** [comment]
Thanks @liuxunorg for your review.

**Pull request closed without merging**